File: 2800-minimum-string-length-after-removing-substrings/2800-minimum-string-length-after-removing-substrings.swift
class Solution {
    func minLength(_ s: String) -> Int {
        var s = s;
        while s.contains("AB") || s.contains("CD") {
            if let range = s.range(of: "AB") {
                s.removeSubrange(range);
            } else if let range = s.range(of: "CD") {
                s.removeSubrange(range);
            }
        }
        return s.count;
    }
}

File: 2800-minimum-string-length-after-removing-substrings/2800-minimum-string-length-after-removing-substrings.rs
impl Solution {
    pub fn min_length(mut s: String) -> i32 {
        while s.contains("AB") || s.contains("CD"){
            if let Some(pos) = s.find("AB"){
                s.replace_range(pos..pos + 2 , "");
            } else if let Some(pos) = s.find("CD"){
                s.replace_range(pos..pos + 2 , "");
            } 
        }
        s.len() as i32
    }
}